### DISCUSSION TITLE
Example Trade Dangerous in Google Colabs

### DISCUSSION THREAD

**Discussion by jcc10** [eyeonus/Trade-Dangerous#190] [Show and tell]
The following is a example of running trade dangerous in google colabs, it is useful if you have limited hardware to run TradeDangerous on (Eg: Steam Deck + Chromebook on a vacation or work trip)

Currently for me (I'm using the free tier) colab python3 runner offers just over 12GB of ram and a little over 100GB of temporary disk storage. (Runners auto-reset after a period of time)

However, I am currently hitting a error when trying to run the `run` command where after a cache rebuild it complains about duplicate stations. (`/content/data/TradeDangerous.prices:5348333 ERROR Second occurrance of station "WAT YU/STRONGHOLD CARRIER", previous entry at line 5341442.`) Said error doesn't seem to happen on my laptop so I'm not sure what's causing it but if anyone else has an idea on how to fix it, I'll update the example notebook.

https://colab.research.google.com/drive/13Mo2Men-q_08Ng92TB2V2BLV5ucPeboh?usp=sharing

EDIT: Was a idiot and forgot the link.

**Comment by eyeonus** [comment]
It is one of the unfortunate limitations of the current specification for .prices files that it does not include a station's unique number and instead uses the name of the station, which is assumed to be unique. The database itself does use the station_id number, so it is capable of handling multiple stations having the same name.

Sometimes the Spansh data that the server uses as a daily baseline has a ship that moved systems as in both the old system and the new. This is the most likely reason for that station being a duplicate.

Simply deleting the TradeDangerous.prices file before running TD again should fix the issue for you, but if it doesn't, you'll need to open the database and remove the duplicate stations as well as deleting the .prices file.
